content/blog/can-we-teach-software-design/index.md

```diff
@@ -0,0 +1,65 @@
+
+### How to Play
+This is mainly about mechanics - building a library of motions (e.g. grooves, fills) and getting comfortable with executing them (e.g. technique, building muscle memory, playing up to speed). These "building blocks" can then be played in sequence to a song.
+
+### When to Play
+This is about knowing the "right" sequence of building blocks to play e.g. does the groove fit this song? Is this the right time to play a particular fill? What sort of fill should I play?
+
+There are principles and even some heuristics, such as:
+- "play a groove for 3 bars, do a fill on the 4th"
+- a mapping of grooves to genres (e.g. you'd often hear the [one-drop](https://en.wikipedia.org/wiki/One_drop_rhythm) in reggae music)
+
+They're important, but knowing these principles and heuristics don't immediately translate to "having great feel". Instead I believe the secret sauce is this: **copying how other drummers play in existing music**.
+
+Copying isn't just about learning "how to play" patterns in the song - as we play to a song, we are also listening to what other musicians are doing, subconsciously connecting drum patterns to things like the melody, intensity of the moment etc. When we next hear a similar type of song, we instinctively know what patterns would likely fit, even if we've never heard the drum parts before.
+
+"Experience" and more free-form practice is still a key part of learning, but I believe copying helps quickly build "feel" for what works, reducing the amount of trial-and-error needed.
+
+## What are we missing in Software Design?
+Here are similarities I found in practicing software design:
+- **"How to play"**:
+   - Design Patterns, Refactoring Techniques
+- **"When to play"**:
+  - Principles and Heuristics (SOLID, DRY, [information hiding](https://wasabigeek.com/blog/what-does-a-1972-paper-have-to-do-with-the-single-responsibility-principle/), abstraction etc.)
+  - Trial-and-error via design katas and actual coding
+  - Code Reviews
+  - **Reading and/or copying code** (e.g. open source)
+
+On the surface, reading and copying code seems like a parallel to copying music, but I feel there's a deficiency - we can generally tell what "good" music is, whether instinctively or by looking at genre top charts.
+
+I feel we don't really have the same for software design (or if we do, it's not well-known).
+
+Maybe it's because not everyone agrees on what constitutes a good design (we tend to argue using principles, which inherently require additional nuance), but I think it's also because while there are foundational aspects of a good design, a sizable part of it contextual. Meaning unless one understands the context behind a design choice, copying blindly is basically trial-and-error (inefficient).
+
+The contextual element is also tricky given that a key goal of design is to reduce cost over time i.e. we can only really tell if a design was good in retrospect.
```

Comment:
Reading this, if I understood correctly, you wanted to convey that what is missing in Software Design is the lack of standards of what constitutes good software design. Maybe you can start with that main point first before talking about the similarities of software design to playing drums. You can also elaborate on more on the lack of standards, probably needs more research.